playbooks/setup_jenkins_slave.yml

```diff
@@ -9,13 +9,61 @@
         state: installed
         update_cache: yes
       with_items:
-        - git
+        - git-core
         - default-jre-headless
         - python-dev
-        - python-pip
-        - build-essential
+        - gcc
+        - libffi-dev
         - libssl-dev
 
+    - block:
+        - name: Get Modern PIP
+          get_url:
+            url: "https://bootstrap.pypa.io/get-pip.py"
+            dest: "/opt/get-pip.py"
+            force: "yes"
+          register: get_pip
+          until: get_pip | success
+          retries: 5
+          delay: 2
+          tags:
+            - pip-install-script
+
+      rescue:
+        - name: Get Modern PIP using fallback URL
+          get_url:
+            url: "https://raw.githubusercontent.com/pypa/get-pip/master/get-pip.py"
+            dest: "/opt/get-pip.py"
+            force: "yes"
+          when: get_pip | failed
+          register: get_pip_fallback
+          until: get_pip_fallback | success
+          retries: 5
+          delay: 2
+          tags:
+            - pip-install-script
+
+
+    - block:
+        - name: Install PIP
+          command: >
+            python /opt/get-pip.py
```

Comment:
You're missing the ``pip setuptools wheel -c rpc-gating/constraints.txt`` parameters.